docs/path.md

```diff
@@ -2,6 +2,22 @@
 
 Since there does not exist a standard for path syntax, RedisJSON implements its own. RedisJSON's syntax is a subset of common best practices and resembles [JSONPath](http://goessner.net/articles/JsonPath/) not by accident.
 
+There is currently two concurrent implementations. One is a legacy from the first version of RedisJSON. Describe below as the `legacy path syntax`.
+
+RedisJSON decide which implementation to used depending on the first character of the path query. If the query starts with the character `$` it is considered as a JSONPath query. Otherwise is is interpreted as a legacy path syntax.
+
+## JSONPath support (RedisJSON v2)
+
+RedisJSON 2.0 introduces the support of [JSONPath](http://goessner.net/articles/JsonPath/). It follows the syntax described by Goessner in his article.
+
+JSONPath query can resolve to several location in the JSON documents. In this case, the JSON commands are applying the operation to every possible location. This is a major improvement over the legacy support, which was operating only on the first path.
+
+Notice that the structure of the command response is most of the time different when using JSONPath. This two behavior are described in the [Commands](/commands) page.
+
```

Comment:
Do we want to mention the limitation of using colon (`:`) character in a path using the dot notation (MOD-1635) and the workaround of using bracket notation?

Reply:
Brilliant !